src/client/pages/Geo/GeoMap/GeoMapMenuData/MapVisualizerPanel/mapVisualizerPanel.tsx

```diff
@@ -1,35 +1,103 @@
 import './mapVisualizerPanel.scss'
-import React from 'react'
+import React, { useCallback } from 'react'
+
+// @ts-ignore
+import ee from '@google/earthengine'
+import axios from 'axios'
+
+import { ForestSources } from '@meta/geo'
+
+import { useAppDispatch } from '@client/store'
+import { GeoActions, useForestSourceOptions } from '@client/store/ui/geo'
+import { useGeoMap } from '@client/hooks'
 
 import GeoMapMenuListElement from '../../GeoMapMenuListElement'
 
 // Just to display items for demo purposes
 const layers = [
-  { title: 'Copernicus (2019)' },
-  { title: 'TanDEM-X (2019)' },
-  { title: 'Hansen 10% (2020)' },
-  { title: 'GlobeLand (2010)' },
+  { key: 'JAXA', title: 'JAXA (2017)', apiUri: '/api/geo/layers/forest/FIN/JAXA/' },
+  { key: 'TandemX', title: 'TanDEM-X (2019)', apiUri: '/api/geo/layers/forest/FIN/TandemX/' },
+  { key: 'GlobeLand', title: 'GloabeLand (2020)', apiUri: '/api/geo/layers/forest/FIN/GlobeLand/' },
+  { key: 'ESAGlobCover', title: 'Global Land Cover ESA (2009)', apiUri: '/api/geo/layers/forest/FIN/ESAGlobCover/' },
+  { key: 'Copernicus', title: 'Copernicus (2019)', apiUri: '/api/geo/layers/forest/FIN/Copernicus/' },
+  { key: 'ESRI', title: 'ESRI (2020)', apiUri: '/api/geo/layers/forest/FIN/ESRI/' },
+  { key: 'ESAWorldCover', title: 'ESA (2020)', apiUri: '/api/geo/layers/forest/FIN/ESAWorldCover/' },
+  { key: 'Hansen', title: 'Hansen GFC (2020)', apiUri: '/api/geo/layers/forest/FIN/Hansen/10/' },
 ]
 
-const SatelliteSourcePanel: React.FC = () => {
+const queryOverlayLayer = (mapLayerId: string, overlayLayers: google.maps.MVCArray) => {
+  for (let i = 0; i < overlayLayers.getLength(); i += 1) {
+    const overlayLayer = overlayLayers.getAt(i)
+    if (overlayLayer.name === mapLayerId) {
+      return true
+    }
+  }
+  return false
+}
+
+const removeOverlayLayer = (mapLayerId: string, overlayLayers: google.maps.MVCArray) => {
+  for (let i = 0; i < overlayLayers.getLength(); i += 1) {
+    const overlayLayer = overlayLayers.getAt(i)
```

Comment:
I saw this loop twice.
maybe better a common method
```
const overlayLayer = findLayer(map,mapLayerId) // I would pass the map reference, not jus the overlayLayers
```

Reply:
Yep, noticed it too and made a mental note to optimize, but there's so many mental notes. 
I will certainly refactor a thing or two by next time. I had a goal to make it work, and I hope I didn't cut too many corners in the effort